doc/build-instructions/Build_Instructions_V11.md

```diff
@@ -784,3 +785,345 @@ JCL      - d247952 based on jdk-11.0.6+6)
+- [bootJDK_OpenJ9](https://adoptopenjdk.net/archive.html?variant=openjdk11&jvmVariant=openj9), which is only used in the cross-compilation on the host system
+- [QEMU](https://www.qemu.org/download/#linux), which is an open source emulator that converts the RISC-V instructions to the opcode on the host system
+- [Fedora_RISCV](https://fedorapeople.org/groups/risc-v/disk-images/), which contains an Fedora OS image and the corresponding BBL (Berkeley Boot Loader) binary file (used to boot via QEMU)
+- [Freemarker V2.3.8](https://sourceforge.net/projects/freemarker/files/freemarker/2.3.8/freemarker-2.3.8.tar.gz/download)
+
+The following sections cover the whole installation & building steps for all required software packages.
+
+### 2. Install software packages for the cross-compilation
+:rocket:
+For the cross-compilation, the first thing is to install all required software packages on your system before building the JDK. When it comes to Fedora, you need to boot the OS image via QEMU to login the target system for installation.
+
+[1] First of all, download and install QEMU on your system as follows:
+```
+sudo apt-get install qemu
+```
+
+:bulb:
+If you are interested in building the QEMU source on your own, please refer to the building instructions
+at https://github.com/riscv/riscv-qemu/wiki for details.
+e.g.
+```
+git clone --recursive https://github.com/riscv/riscv-qemu.git
+cd riscv-qemu
+./configure  --target-list=riscv64-softmmu,riscv64-linux-user
+make -j$(nproc)
+make install
+```
+
+[2] Install `e2fsck` if it doesn't exist on your host system:
+```
+wget http://downloads.sourceforge.net/project/e2fsprogs/e2fsprogs/v1.43.1/e2fsprogs-1.43.1.tar.gz
+tar xzf e2fsprogs-1.43.1.tar.gz
+cd e2fsprogs-1.43.1
+./configure # <== If this step fails, please check the config.log file.
+                  It could be the missing "libc6-dev" package on your system.
+make
+make install
+```
+
+[3] Download the Fedora OS image `stage4-disk.img.xz` and `bbl` (the boot loader) from
+https://fedorapeople.org/groups/risc-v/disk-images/ (4GB in size by default)
+and expand it appropriately (e.g. 20GB) for more space as follows:
+```
+unxz --keep stage4-disk.img.xz
+truncate -s 20G stage4-disk.img
+e2fsck -fp stage4-disk.img
+resize2fs stage4-disk.img
+```
+
+[4] Boot the Fedora OS via QEMU (please refer to https://wiki.qemu.org/Documentation/Platforms/RISCV for details).
+e.g.
+```
+  qemu-system-riscv64 \
+   -nographic \
+   -machine virt \
+   -smp 4 \
+   -m 2G  \   # the memory can be adjusted depending on the capability of your host system
+   -kernel bbl \
+   -object rng-random,filename=/dev/urandom,id=rng0 \
+   -device virtio-rng-device,rng=rng0 \
+   -append "console=ttyS0 ro root=/dev/vda" \
+   -device virtio-blk-device,drive=hd0 \
+   -drive file=stage4-disk.img,format=raw,id=hd0 \
+   -device virtio-net-device,netdev=usernet \
+   -netdev user,id=usernet,hostfwd=tcp::10000-:22 (the port number is 10000)
```

Comment:
Please use `#` for comments consistently.

Reply:
Fixed as suggested.